File: readability.c
// Program to find the grade of a specific text;

#include <ctype.h>// how many letter, count
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    float L = (letters / words)*100;
    float S = (sentences / words)*100;

    // Compute the Coleman-Liau index
    float score = (0.0588 * L - 0.296 * S - 15.8);

    // Print the grade level
    if (score < 1.0)
    {
        printf("Before Grade 1\n");
    }
    else if (1.0 <= score && score < 16.0)
    {
        //round
        printf("Grade %d\n", (int) round(score));
    }
    else
    {
        printf("Grade 16+\n");
    }
}

int count_letters(string text)
{
    // Return the number of letters in text
    int length = strlen(text);
    int count = 0;

    for (int i = 0; text[i] != '\0'; i++)
        if (text[i] == ' ')
            count++;
        else if (text[i] == '.')
            count++;
        else if (text[i] == ',')
            count++;
        else if (text[i] == ':')
            count++;
        else if (text[i] == ';')
            count++;
        else if (text[i] == '?')
            count++;
        else if (text[i] == '!')
            count++;
        else if (text[i] == '\'')
            count++;
        else if (text[i] == '\"')
            count++;
    return length - count;
}



int count_words(string text)
{
    // Return the number of words in text

    int count = 0;

    for (int i = 0; text[i] != '\0'; i++)
        if (text[i] == ' ')
            count++;
    return count + 1;

}

int count_sentences(string text)
{
    // Return the number of sentences in text
    int count = 0;

    for (int i = 0; text[i] != '\0'; i++)
        if (text[i] == '!')
            count++;
        else if (text[i] == '?')
            count++;
        else if (text[i] == '.')
            count++;
    return count;

}

File: readability.py
###from cs50 import get_string

try:
    text = str(input("Text: "))
except ValueError:
    exit(1)

len_text = len(text)
words = 1
result = 0
letters = 0
phrases = 0

for i in range(len_text):
    if text[i].isalpha() or "-" in text[i]:
       letters += 1
    else:
        if " " in text[i]:
            words += 1
        else:
            for char in {"?", "!", "."}:
                if char in text[i]:
                    phrases += 1

S = (phrases / words) * 100

L = (letters / words) * 100

result = 0.0588 * L - 0.296 * S - 15.8

result = round(result)
####print(result)

if result < 1:
    print("Before Grade 1")
    exit(0)

elif result < 16:
    print("Grade ", result)

else:
    print("Grade 16+")


